{"expand":"renderedFields,names,schema,transitions,operations,editmeta,changelog","id":"12406442","self":"https://issues.apache.org/jira/rest/api/latest/issue/12406442","key":"DERBY-3913","fields":{"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/1","id":"1","description":"A problem which impairs or prevents the functions of the product.","iconUrl":"https://issues.apache.org/jira/images/icons/issuetypes/bug.png","name":"Bug","subtask":false},"timespent":null,"project":{"self":"https://issues.apache.org/jira/rest/api/2/project/10594","id":"10594","key":"DERBY","name":"Derby","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/projectavatar?pid=10594&avatarId=10122","24x24":"https://issues.apache.org/jira/secure/projectavatar?size=small&pid=10594&avatarId=10122","16x16":"https://issues.apache.org/jira/secure/projectavatar?size=xsmall&pid=10594&avatarId=10122","32x32":"https://issues.apache.org/jira/secure/projectavatar?size=medium&pid=10594&avatarId=10122"},"projectCategory":{"self":"https://issues.apache.org/jira/rest/api/2/projectCategory/10090","id":"10090","description":"DB related projects","name":"DB"}},"fixVersions":[{"self":"https://issues.apache.org/jira/rest/api/2/version/12316344","id":"12316344","description":"First release on the 10.9 branch","name":"10.9.1.0","archived":false,"released":true,"releaseDate":"2012-06-25"}],"aggregatetimespent":null,"resolution":{"self":"https://issues.apache.org/jira/rest/api/2/resolution/1","id":"1","description":"A fix for this issue is checked into the tree and tested.","name":"Fixed"},"customfield_12310220":"2009-07-03 11:49:27.938","customfield_12312322":null,"customfield_12312323":null,"customfield_12310420":"23915","customfield_12310222":"1_*:*_1_*:*_84253814200_*|*_5_*:*_1_*:*_0","customfield_12312320":null,"customfield_12312321":null,"customfield_12312120":null,"customfield_12312121":null,"resolutiondate":"2011-06-17T00:56:39.501+0000","workratio":-1,"customfield_12312328":null,"customfield_12312329":null,"customfield_12312326":null,"customfield_12310300":null,"customfield_12312327":null,"customfield_12312324":null,"customfield_12312325":null,"lastViewed":null,"watches":{"self":"https://issues.apache.org/jira/rest/api/2/issue/DERBY-3913/watchers","watchCount":1,"isWatching":false},"created":"2008-10-14T21:06:25.352+0000","priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/5","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/trivial.png","name":"Trivial","id":"5"},"labels":["derby_backport_reject_10_8","derby_triage10_5_2"],"customfield_12312333":null,"customfield_12312334":null,"customfield_12310310":"1.0","customfield_12312331":null,"customfield_12312332":null,"timeestimate":null,"aggregatetimeoriginalestimate":null,"versions":[{"self":"https://issues.apache.org/jira/rest/api/2/version/12313771","id":"12313771","description":"build: 764942, 14/Apr/09. Voted release: 28/Apr/09. Announced: 1/May/09","name":"10.5.1.1","archived":false,"released":true,"releaseDate":"2009-04-28"}],"customfield_12311120":null,"customfield_12312330":null,"issuelinks":[{"id":"12339804","self":"https://issues.apache.org/jira/rest/api/2/issueLink/12339804","type":{"id":"10030","name":"Reference","inward":"is related to","outward":"relates to","self":"https://issues.apache.org/jira/rest/api/2/issueLinkType/10030"},"inwardIssue":{"id":"12409692","key":"DERBY-3970","self":"https://issues.apache.org/jira/rest/api/2/issue/12409692","fields":{"summary":"PositionedStoreStream doesn't initialize itself properly","status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Complete"}},"priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/4","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/minor.png","name":"Minor","id":"4"},"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/1","id":"1","description":"A problem which impairs or prevents the functions of the product.","iconUrl":"https://issues.apache.org/jira/images/icons/issuetypes/bug.png","name":"Bug","subtask":false}}}}],"customfield_12312339":null,"assignee":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mamtas","name":"mamtas","emailAddress":"msatoor at gmail dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Mamta A. Satoor","active":true},"customfield_12312337":null,"customfield_12312338":null,"updated":"2013-06-17T09:19:29.489+0000","customfield_12312335":null,"customfield_12311720":null,"customfield_12312336":null,"status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Complete"}},"components":[{"self":"https://issues.apache.org/jira/rest/api/2/component/11400","id":"11400","name":"Miscellaneous"}],"timeoriginalestimate":null,"description":"I found a script, trunk/tools/testing/i18nTestGenerator/generateClientMessageTest.sh, intended to create a test to verify correctness of client error messages(trunk/java/testing/org/apache/derbyTesting/functionTests/tests/i18n/TestClientMessages.java ). \r\nThe script is broken (see DERBY-1567) but after fixing up the resulting test and running it, it did show two messages which look a little odd in their usage (plus some messages for which the usage looked fine):\r\n                                                            \r\nXCL30 - LANG_STREAMING_COLUMN_I_O_EXCEPTION: \r\nmessages.xml: \r\n            <msg>\r\n                <name>XCL30.S</name>\r\n                <text>An IOException was thrown when reading a '{0}' from an InputStream.</text>\r\n                <arg>value</arg>\r\n            </msg>\r\napparently correct number of parameters, but odd...doesn't look like ioe fits the usage in the message text.\r\nEmbedBlob:\r\n            } catch (IOException ioe) {\r\n                throw StandardException.newException(\r\n                        SQLState.LANG_STREAMING_COLUMN_I_O_EXCEPTION, ioe);\r\n            }\r\nhas a second parameter:\r\nclient.am.Lob:\r\n            throw new SqlException(null,\r\n                        new ClientMessageId(\r\n                            SQLState.LANG_STREAMING_COLUMN_I_O_EXCEPTION),\r\n                        typeDesc,\r\n                        ioe\r\n                    );\r\nlooks like second parameter fits the {0}:\r\nSQLBinary: \r\n\t\tthrow StandardException.\r\n\t\t\tnewException(SQLState.LANG_STREAMING_COLUMN_I_O_EXCEPTION,\r\n\t\t\t\t\t\t ioe, getTypeName());\r\nSQLChar:\r\n\t\tthrow StandardException.\r\n\t\t\tnewException(SQLState.LANG_STREAMING_COLUMN_I_O_EXCEPTION,\r\n\t\t\t\t\t\t ioe, getTypeName());\r\n\r\n                   throw StandardException.newException(\r\n                            SQLState.LANG_STREAMING_COLUMN_I_O_EXCEPTION, \r\n                            ioe, \r\n                            \"java.sql.String\");\r\n--------------------------------------------------------------\r\n22003.S.4 - CLIENT_LENGTH_OUTSIDE_RANGE_FOR_DATATYPE\r\n            <msg>\r\n                <name>22003.S.4</name>\r\n                <text>The length ({0}) exceeds the maximum length for the data type ({1}).</text>\r\n                <arg>number</arg>\r\n                <arg>datatypeName</arg>\r\n            </msg>\r\n\r\ncorrect number of parameters, but new Integer(Integer.MAX_VALUE) returns a number, not a datatype name:             \r\nclient.am.PreparedStatement:\r\n\t\tthrow new SqlException(\r\n                        agent_.logWriter_,\r\n                        new ClientMessageId(\r\n                            SQLState.CLIENT_LENGTH_OUTSIDE_RANGE_FOR_DATATYPE),\r\n                        new Long(length),\r\n                        new Integer(Integer.MAX_VALUE)\r\n                    ).getSQLException();\r\nclient.am.ResultSet:\r\n                throw new SqlException(agent_.logWriter_,\r\n                    new ClientMessageId(SQLState.CLIENT_LENGTH_OUTSIDE_RANGE_FOR_DATATYPE),\r\n                    new Long(length), new Integer(Integer.MAX_VALUE)).getSQLException();\r\n                throw new SqlException(agent_.logWriter_,\r\n                    new ClientMessageId(SQLState.CLIENT_LENGTH_OUTSIDE_RANGE_FOR_DATATYPE),\r\n                    new Long(length), new Integer(Integer.MAX_VALUE)).getSQLException();                \r\n\t\tthrow new SqlException(agent_.logWriter_,\r\n                    new ClientMessageId(SQLState.CLIENT_LENGTH_OUTSIDE_RANGE_FOR_DATATYPE),\r\n                    new Long(length), new Integer(Integer.MAX_VALUE)).getSQLException();\r\n\r\n-------------------------------------------","customfield_10010":null,"timetracking":{},"customfield_12312026":null,"customfield_12310200":null,"customfield_12312023":null,"customfield_12312024":null,"customfield_12312340":null,"aggregatetimeestimate":null,"customfield_12312022":null,"customfield_12312341":null,"customfield_12312220":null,"customfield_12310921":null,"customfield_12310920":"36407","summary":"mismatch between error XCL30 and 22003.S.4 and parameters in usage","creator":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=myrna","name":"myrna","emailAddress":"m dot v dot lunteren at gmail dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Myrna van Lunteren","active":true},"subtasks":[],"customfield_12310090":[{"self":"https://issues.apache.org/jira/rest/api/2/customFieldOption/10423","value":"Newcomer","id":"10423"}],"reporter":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=myrna","name":"myrna","emailAddress":"m dot v dot lunteren at gmail dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Myrna van Lunteren","active":true},"customfield_12310291":null,"customfield_12310290":null,"aggregateprogress":{"progress":0,"total":0},"environment":null,"customfield_12311020":null,"customfield_12310050":{"self":"https://issues.apache.org/jira/rest/api/2/customFieldOption/10052","value":"Normal","id":"10052"},"duedate":null,"customfield_12310250":null,"progress":{"progress":0,"total":0},"comment":{"startAt":0,"maxResults":5,"total":5,"comments":[{"self":"https://issues.apache.org/jira/rest/api/2/issue/12406442/comment/12726917","id":"12726917","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kristwaa","name":"kristwaa","emailAddress":"kristwaa at apache dot org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=kristwaa&avatarId=14736","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=kristwaa&avatarId=14736","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=kristwaa&avatarId=14736","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=kristwaa&avatarId=14736"},"displayName":"Kristian Waagan","active":true},"body":"Triaged July 3, 2009: Assigned normal urgency.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kristwaa","name":"kristwaa","emailAddress":"kristwaa at apache dot org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=kristwaa&avatarId=14736","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=kristwaa&avatarId=14736","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=kristwaa&avatarId=14736","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=kristwaa&avatarId=14736"},"displayName":"Kristian Waagan","active":true},"created":"2009-07-03T11:49:27.938+0000","updated":"2009-07-03T11:49:27.938+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12406442/comment/13048985","id":"13048985","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mamtas","name":"mamtas","emailAddress":"msatoor at gmail dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Mamta A. Satoor","active":true},"body":"The script generateClientMessageTest.sh which is supposed to generate a test to verify the correctness of the error messages is preety badly broken.I tried fixing the errors in the generated test but there were just too many problems in the generated test. \r\n\r\nI have checked in the fixes for the mismatch between error XCL30 and 22003.S.4 and parameters in usage though and hence this jira can be closed. There is another jira DERBY-1567 to fix the script when generates the test.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mamtas","name":"mamtas","emailAddress":"msatoor at gmail dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Mamta A. Satoor","active":true},"created":"2011-06-14T05:36:02.303+0000","updated":"2011-06-14T05:36:02.303+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12406442/comment/13049055","id":"13049055","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"The problem with XCL30.S seems to have been fixed in DERBY-3970, so now all exceptions with that message id are passed one argument and linked to the underlying IOException.\r\n\r\nThe fix committed here seems to have broken this in client.am.Lob. By changing the order of the arguments, we now pass two message arguments, and the generated SqlException is not linked to the underlying IOException. I suppose the confusion comes from the different signatures for the methods that create exceptions in the client and in the engine. In the engine, the cause should be passed before the message arguments, whereas it should be the last argument in the client.\r\n\r\nThe attached patch reverts that part of the fix for this issue.\r\n\r\nCommitted revision 1135404.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2011-06-14T08:19:43.077+0000","updated":"2011-06-14T08:19:43.077+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12406442/comment/13057373","id":"13057373","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=myrna","name":"myrna","emailAddress":"m dot v dot lunteren at gmail dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Myrna van Lunteren","active":true},"body":"Added a reject backport 10.8 because the remaining change changed the construction of the message, and the translations were recently contributed for 10.8.\r\nNormally when we change a message we ought to remove it from the non-English locales (until a new translation) so the English gets picked up at run-time, but I don't think it's necessary in this case, because the behavior for those other languages remains the same as it was before the change. ","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=myrna","name":"myrna","emailAddress":"m dot v dot lunteren at gmail dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Myrna van Lunteren","active":true},"created":"2011-06-29T18:02:13.596+0000","updated":"2011-06-29T18:16:40.620+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12406442/comment/13685257","id":"13685257","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"[bulk update] Close all resolved issues that haven't been updated for more than one year.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2013-06-17T09:19:29.478+0000","updated":"2013-06-17T09:19:29.478+0000"}]},"votes":{"self":"https://issues.apache.org/jira/rest/api/2/issue/DERBY-3913/votes","votes":0,"hasVoted":false},"worklog":{"startAt":0,"maxResults":20,"total":0,"worklogs":[]},"customfield_12311820":"0|i06lfr:"}}